mailscanner/functions.php

```diff
@@ -461,7 +461,7 @@ function printServiceStatus()
         // is MTA running
         $mta = get_conf_var('mta');
         if ('msmail' === $mta) {
-            exec('ps ax | grep postfix | grep -v grep | grep -v php', $output);
+            exec('ps -U postfix -u postfix | grep -v MailScanner | grep -v "MailWatch SQL" | grep -v MSMilter | grep -v PID', $output);
```

Comment:
How about this?

```
ps --pid `ps -C master -o pid=` --ppid `ps -C master -o pid=` -o pid=,cmd=
```

This would search for the postfix/master proc, lookup its childs and output both.

sample output:
```
1601555 /usr/lib/postfix/sbin/master -w
1601557 qmgr -l -t fifo -u
1601721 tlsmgr -l -t unix -u -c
2052128 pickup -l -t fifo -u -c -o content_filter= -o receive_override_options=no_header_body_checks
2066535 showq -t unix -u -c
```

I'm not sure if all used ps parameters are also available on BSD. Can someone confirm?

Reply:
How about something like this?

```
            exec('ps -C master -o pid=',$masterpid);
            if (count($masterpid) > 0) {
                $running = $yes;
                exec('ps --pid '.$masterpid[0].' --ppid '.$masterpid[0].' -o pid=',$childpids);
                $procs = count($childpids) . ' ' . __('procs03');
            } else {
                $running = $no;
                $procs = '0 ' . __('procs03');
            }
```

This would search for the postfix/master proc, lookup its childs and output the total count of both.

I'm not sure if all used ps parameters are also available on BSD. Can someone confirm?